azurerm/resource_arm_storage_table_entity_test.go

```diff
@@ -0,0 +1,275 @@
+		partitionKey := rs.Primary.Attributes["partition_key"]
+		rowKey := rs.Primary.Attributes["row_key"]
+
+		storageClient := testAccProvider.Meta().(*ArmClient).storage
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		resourceGroup, err := storageClient.FindResourceGroup(ctx, accountName)
+		if err != nil {
+			return fmt.Errorf("Error finding Resource Group: %s", err)
+		}
+
+		client, err := storageClient.TableEntityClient(ctx, *resourceGroup, accountName)
+		if err != nil {
+			return fmt.Errorf("Error building Table Entity Client: %s", err)
+		}
+
+		input := entities.GetEntityInput{
+			PartitionKey:  partitionKey,
+			RowKey:        rowKey,
+			MetaDataLevel: entities.NoMetaData,
+		}
+		resp, err := client.Get(ctx, accountName, tableName, input)
+		if err != nil {
+			return fmt.Errorf("Bad: Get on Table EntityClient: %+v", err)
+		}
+
+		if resp.StatusCode == http.StatusNotFound {
+			return fmt.Errorf("Bad: Entity (Partition Key %q / Row Key %q) (Table %q / Account %q / Resource Group %q) does not exist", partitionKey, rowKey, tableName, accountName, *resourceGroup)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMTableEntityDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_storage_table_entity" {
+			continue
+		}
+
+		tableName := rs.Primary.Attributes["table_name"]
+		accountName := rs.Primary.Attributes["storage_account_name"]
+		partitionKey := rs.Primary.Attributes["parititon_key"]
+		rowKey := rs.Primary.Attributes["row_key"]
+
+		storageClient := testAccProvider.Meta().(*ArmClient).storage
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		resourceGroup, err := storageClient.FindResourceGroup(ctx, accountName)
+		if err != nil {
+			return fmt.Errorf("Error finding Resource Group: %s", err)
+		}
+
+		// not found, the account's gone
+		if resourceGroup == nil {
+			return nil
+		}
+
+		client, err := storageClient.TableEntityClient(ctx, *resourceGroup, accountName)
+		if err != nil {
+			return fmt.Errorf("Error building TableEntity Client: %s", err)
+		}
+
+		input := entities.GetEntityInput{
+			PartitionKey: partitionKey,
+			RowKey:       rowKey,
+		}
+		resp, err := client.Get(ctx, accountName, tableName, input)
+		if err != nil {
+			return fmt.Errorf("Bad: Get on Table Entity: %+v", err)
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("Table Entity still exists:\n%#v", resp)
+		}
+	}
+
+	return nil
+}
+
+func testAccAzureRMTableEntity_basic(rInt int, rString string, location string) string {
+	template := testAccAzureRMTableEntity_template(rInt, rString, location)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_storage_table_entity" "test" {
+  storage_account_name = "${azurerm_storage_account.test.name}"
+  table_name           = "${azurerm_storage_table.test.name}"
+  
+  partition_key = "test_partition%d"
+  row_key       = "test_row%d"
+  entity = {
+    foo = "bar"
```

Comment:
Should we check caps here:
```suggestion
    Foo = "Bar"
```